src/assets/Generator.Shared/Utf8JsonWriterExtensions.cs

```diff
@@ -140,5 +141,53 @@ public static void WriteObjectValue(this Utf8JsonWriter writer, object value)
                     throw new NotSupportedException("Not supported type " + value.GetType());
             }
         }
+
+        public static void WriteBinaryDataValue(this Utf8JsonWriter writer, BinaryData value)
+        {
+            byte[] inputValue = value.ToArray();
+
+            // We will remove the trailing occurrences of the specified characters including: TAB, LF, space.
+            byte[] trimBytes = { 9, 10, 32 };
+            int end = inputValue.Length - 1;
+            for (; end >= 0; end--)
+            {
+                int i = 0;
+                for (; i < trimBytes.Length; i++)
+                {
+                    if (trimBytes[i] == inputValue[end])
+                        break;
+                }
+                if (i == trimBytes.Length)
+                    break;
+            }
+
+            // Check if it is possible to serialize the value as json value.
+            // Possible match kinds: Json Object, Json Array, Json value false, Json value true, Json number.
+            // Our logic here is that, for example, we check if the first and the last bytes are '{' and '}', respectively.
+            if (end >= 0 && (
+                inputValue[0] == 123 && inputValue[end] == 125 ||
+                inputValue[0] == 91 && inputValue[end] == 93 ||
+                inputValue[0] == 116 && inputValue[end] == 101 ||
```

Comment:
we are saying that if the raw bytes = 't', ..., 'e' we will treat this as json?  Lets add a test case that sets the bytes equal to `terse`.

I am still not fully sold on the true, false, and number checks here.  If we send them to write as json 456 comes out as "456" according to the test case [here](https://github.com/Azure/autorest.csharp/pull/2544/files#diff-ede2da49b9fb105e853158da8dfe79723aa96156ae93c3577a7b0485fdf24c39R45)?  If we send it to write as string doesn't 456 also become "456"?

If we want to actually do anything here for true / false lets check length then validate the full word.

Reply:
I have added the test case. 

If we write 456 as string, the result will be "\"456\"", not "456". 

I understand the we can check the length and validate the full word for true/false, but what I want to say is, we are using the first and last byte to check if the binary data is likely to be a valid json and we should keep this pattern for the different json kinds. In 90% of the cases, we will pass this check and the json parse will succeed, that's what we are expecting. And we are clear that there are still some cases that the binary data can pass this check but it can't be parsed, therefore we will catch the exception and write it as string.